mitodl/mitxpro
# style: Order receipt design tweaks

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/hq/issues/2953

#### What's this PR do?

Applies the design changes as per mentioned in https://github.com/mitodl/hq/issues/2953

- Adds HSN in the email receipt and the in-app receipt
- Adds GSTIN = `Pending` in email receipt and in-app receipt
- Adds `Total Before Tax` in the email receipt and the in-app receipt
- Changes the Total before tax to `Total Before Tax` on the checkout page
- Adds signature in the in-app receipt
- Changes email receipt design to match as [This figma design](https://www.figma.com/proto/140ipJMKt1up7wHA0OUoXY/xPRO---VAT-Address?page-id=0%3A1&type=design&node-id=277-22&viewport=-2298%2C-482%2C0.35&t=DZbnQFNrbgK4ruNw-1&scaling=min-zoom)
- Makes the base email template configurable as per pre and post logo tax sections


#### How should this be manually tested?
- Setup xPRO
- Setup Courses
- Setup some discounts (i.e. 40% and 100%)
- Add Tax Rate for your IP, To test easily you can set `ECOMMERCE_FORCE_PROFILE_COUNTRY=True` in env file
- Now create a couple of orders e.g. with 100% discount, 40% discount, and no discount
- Place orders and match the in-app receipt and receipt emails. They should match the design
- Also, Place some orders with and without tax applied to see if everything works fine. The Tax & price values on the receipt should be correct
- Open the printable page for the in-app receipt and they should look fine almost like the previous design with additional signature and fields i.e. `Total Before Tax`
- Match the overall design changes
- Since we changed the base email template, It would also be good to check other emails for regressions i.e. enrollment email, change password email, etc.


Additional Test Cases:

Disable `FEATURE_ENABLE_TAXES_DISPLAY=False` And you shouldn't:
1. See HSN number in the in-app receipt or Email
2. See signature in the inline receipt
3. GSTIN in the in-app receipt and email.


#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
![screencapture-xpro-odl-local-8053-receipt-2-2023-12-08-17_46_06](https://github.com/mitodl/mitxpro/assets/34372316/65a887ea-1588-4c6b-ba92-4eb61a6fa670)

<img width="1013" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/0f0144a0-ba47-46c3-9f26-e94fd90bbfc9">

<img width="686" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/4c0bd18c-fdec-443e-a8cc-cb85bd0c30a1">

<img width="1194" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/b658b528-4751-4eac-93e4-a1205753f5d8">


#### Mobile Screenshots

![screencapture-xpro-odl-local-8053-receipt-2-2023-12-08-17_46_16](https://github.com/mitodl/mitxpro/assets/34372316/05807123-0da6-4a45-962d-0c15cc94e1a1)

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/l2JHVUriDGEtWOx0c/giphy.gif)
